addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -136,482 +171,192 @@ static func _category_cmp(a: BlockCategory, b: BlockCategory) -> bool:
-	b.block_name = "viewport_width"
-	b.variant_type = TYPE_FLOAT
-	b.block_format = "Viewport Width"
-	b.statement = "(func (): var transform: Transform2D = get_viewport_transform(); var scale: Vector2 = transform.get_scale(); return -transform.origin.x / scale.x + get_viewport_rect().size.x / scale.x).call()"
-	b.category = "Graphics | Viewport"
-	block_list.append(b)
-
-	b = BLOCKS["parameter_block"].instantiate()
-	b.block_name = "viewport_height"
-	b.variant_type = TYPE_FLOAT
-	b.block_format = "Viewport Height"
-	b.statement = "(func (): var transform: Transform2D = get_viewport_transform(); var scale: Vector2 = transform.get_scale(); return -transform.origin.y / scale.y + get_viewport_rect().size.y / scale.y).call()"
-	b.category = "Graphics | Viewport"
-	block_list.append(b)
-
-	b = BLOCKS["parameter_block"].instantiate()
-	b.block_name = "viewport_center"
-	b.variant_type = TYPE_VECTOR2
-	b.block_format = "Viewport Center"
-	b.statement = "(func (): var transform: Transform2D = get_viewport_transform(); var scale: Vector2 = transform.get_scale(); return -transform.origin / scale + get_viewport_rect().size / scale / 2).call()"
-	b.category = "Graphics | Viewport"
-	block_list.append(b)
 
-#endregion
+static func get_parameter_output_blocks(block_definition_list: Array[BlockDefinition]) -> Array[BlockDefinition]:
+	var param_output_blocks: Array[BlockDefinition] = []
 
-	return block_list
+	for block_definition in block_definition_list:
+		# Block must be entry to have parameter outputs (helps with performance of this method)
+		if block_definition.type != Types.BlockType.ENTRY:
+			continue
+
+		var regex = RegEx.create_from_string("\\[([^\\]]+)\\]")  # Capture things of format [test]
+		var results := regex.search_all(block_definition.display_template)
+
+		for result in results:
+			var param := result.get_string()
+			param = param.substr(1, param.length() - 2)
+			var split := param.split(": ")
+
+			var param_name := split[0]
+			var param_type_str := split[1]
+			var param_type = Types.STRING_TO_VARIANT_TYPE[param_type_str]
 
+			var bd = BlockDefinition.new()
+			bd.name = block_definition.name + "_" + param_name
+			bd.category = block_definition.category
+			bd.type = Types.BlockType.VALUE
+			bd.variant_type = param_type
+			bd.display_template = param_name
+			bd.code_template = param_name
+			bd.scope = block_definition.code_template
 
-static func property_to_blocklist(property: Dictionary) -> Array[Block]:
-	var block_list: Array[Block] = []
+			param_output_blocks.append(bd)
+
+	return param_output_blocks
+
+
+static func property_to_blocklist(property: Dictionary) -> Array[BlockDefinition]:
+	var block_definition_list: Array[BlockDefinition] = []
 
 	var variant_type = property.type
 
 	const FALLBACK_SET_FOR_TYPE = {
-		TYPE_INT: "0",
-		TYPE_FLOAT: "0",
-		TYPE_VECTOR2: "0,0",
-		TYPE_COLOR: "DARK_ORANGE",
+		TYPE_INT: 0,
+		TYPE_FLOAT: 0.0,
+		TYPE_VECTOR2: Vector2(0, 0),
+		TYPE_COLOR: Color.SKY_BLUE,
```

Comment:
You didn't like DARK_ORANGE? :) I though it played nice with Endless colors.

Reply:
Idk why I felt threatened by the orange that day... it looks good to me now haha.